libs/platform/src/lib/components/menu-button/menu-button.component.ts

```diff
@@ -30,10 +39,26 @@ export class MenuButtonComponent extends BaseComponent {
     @Output()
     buttonClicked: EventEmitter<MouseEvent | KeyboardEvent | TouchEvent> = new EventEmitter();
 
+    /**
+     * @hidden disabling fd-button does not disables menu button. because menu trigger is on menu-button; menu gets open.
+     * to prevent this, need to apply is-disabled at menu-button level as well.
+     */
+    @HostBinding('class.is-disabled')
+    menuButtonDisabled = false;
+
+    // tabindex for button.
+    tabindex = 0;
+
     constructor(_cd: ChangeDetectorRef) {
         super(_cd);
     }
 
+    /** @hidden */
+    ngOnInit(): void {
+        this.menuButtonDisabled = this.disabled;
+        this.tabindex = this.disabled ? -1 : 0;
```

Comment:
probably we need a getter for it to keep it up to date cause `disabled` state might happen after `OnInit` phase

Reply:
using getter and setter now.

Reply:
It's not what I meant.
In your case when `disabled` gets changed after OnInit `tabindex` and `menuButtonDisabled` remains the same and doesn't reflect this change. Please correct me if I'm wrong.

What I mean:
```typescript
@HostBinding('class.menu-button-disabled')
get  menuButtonDisabled(): boolean {
    return this.disabled;
}

get tabindex(): number {
    return this.disabled ? -1 : 0;
}
```

Reply:
okay.. understood your point. have corrected the code now.